pvlib/iotools/pvgis.py

```diff
@@ -23,6 +23,348 @@
+    P†                           float   PV system power (W)
+    G(i), poa_global‡            float   Global irradiance on inclined plane (W/m^2)
+    Gb(i), poa_direct‡           float   Beam (direct) irradiance on inclined plane (W/m^2)
+    Gd(i), poa_sky_diffuse‡      float   Diffuse irradiance on inclined plane (W/m^2)
+    Gr(i), poa_ground_diffuse‡   float   Reflected irradiance on inclined plane (W/m^2)
+    H_sun, solar_elevation       float   Sun height/elevation (degrees)
+    T2m, temp_air                float   Air temperature at 2 m (degrees Celsius)
+    WS10m, wind_speed            float   Wind speed at 10 m (m/s)
+    Int                          int     Solar radiation reconstructed (1/0)
+    ===========================  ======  ======================================
+
+    †P (PV system power) is only returned when pvcalculation=True.
+
+    ‡Gb(i), Gd(i), and Gr(i) are returned when components=True, otherwise the
+    sum of the three components, G(i), is returned.
+
+    Raises
+    ------
+    requests.HTTPError
+        if the request response status is ``HTTP/1.1 400 BAD REQUEST``, then
+        the error message in the response will be raised as an exception,
+        otherwise raise whatever ``HTTP/1.1`` error occurred
+
+    See Also
+    --------
+    pvlib.iotools.read_pvgis_hourly, pvlib.iotools.get_pvgis_tmy
+
+    References
+    ----------
+    .. [1] `PVGIS <https://ec.europa.eu/jrc/en/pvgis>`_
+    .. [2] `PVGIS Hourly Radiation
+       <https://ec.europa.eu/jrc/en/PVGIS/tools/hourly-radiation>`_
+    .. [3] `PVGIS Non-interactive service
+       <https://ec.europa.eu/jrc/en/PVGIS/docs/noninteractive>`_
+    .. [4] `PVGIS horizon profile tool
+       <https://ec.europa.eu/jrc/en/PVGIS/tools/horizon>`_
+    """  # noqa: E501
+    # use requests to format the query string by passing params dictionary
+    params = {'lat': latitude, 'lon': longitude, 'outputformat': outputformat,
+              'angle': surface_tilt, 'aspect': surface_azimuth,
+              'pvcalculation': int(pvcalculation),
+              'pvtechchoice': pvtechchoice, 'mountingplace': mountingplace,
+              'trackingtype': trackingtype, 'components': int(components),
+              'usehorizon': int(usehorizon),
+              'optimalangles': int(optimalangles),
+              'optimalinclination': int(optimalangles), 'loss': loss}
+    # pvgis only takes 0 for False, and 1 for True, not strings, also the
+    if userhorizon is not None:
+        params['userhorizon'] = ','.join(str(x) for x in userhorizon)
+    if raddatabase is not None:
+        params['raddatabase'] = raddatabase
+    if (start is not None) & (type(start) is int):
+        params['startyear'] = start
+    elif start is not None:
+        params['startyear'] = start if isinstance(start, int) else start.year
+    if end is not None:
+        params['endyear'] = end if isinstance(end, int) else end.year
```

Comment:
If you were to pass a float right now it would fail because line 190 would attempt to do start.year which throw an error. 

I'm not quite sure what you are suggesting? Are you suggesting also allowing floats, e.g.: 

    params['startyear'] = start if (isinstance(start, int) | isinstance(start, float)) else start.year

Reply:
There's not a clean 1-liner for what I was suggesting, so let's ignore. Sorry for the noise. FYI `isinstance` can take a tuple of types e.g. `isinstance(start, (int, float))`.